docs/core/extensions/artificial-intelligence.md

```diff
@@ -0,0 +1,632 @@
+
+- <xref:Microsoft.Extensions.AI.AIFunction>: Represents a function that can be described to an AI service and invoked.
+- <xref:Microsoft.Extensions.AI.AIFunctionFactory>: Provides factory methods for creating commonly used implementations of `AIFunction`.
+- <xref:Microsoft.Extensions.AI.FunctionInvokingChatClient>: Wraps an `IChatClient` to add automatic function invocation capabilities.
+
+Consider the following example, that demonstrates a random function invocation:
+
+```csharp
+using System.ComponentModel;
+using Microsoft.Extensions.AI;
+
+[Description("Gets the current weather")]
+string GetCurrentWeather() => Random.Shared.NextDouble() > 0.5
+    ? "It's sunny"
+    : "It's raining";
+
+IChatClient client = new ChatClientBuilder(
+        new OllamaChatClient(new Uri("http://localhost:11434"), "llama3.1"))
+    .UseFunctionInvocation()
+    .Build();
+
+var response = client.CompleteStreamingAsync(
+    "Should I wear a rain coat?",
+    new() { Tools = [ AIFunctionFactory.Create(GetCurrentWeather) ] });
+
+await foreach (var update in response)
+{
+    Console.Write(update);
+}
+```
+
+The preceding code:
+
+- Defines a function named `GetCurrentWeather` that returns a random weather forecast.
+  - This function is decorated with a `Description` attribute, which is used to provide a description of the function to the AI service.
+- Instantiates a `ChatClientBuilder` with an `OllamaChatClient` and configures it to use function invocation.
+- Calls `CompleteStreamingAsync` on the client, passing a prompt and a list of tools that includes a function created with `AIFunctionFactory.Create`.
+- Iterates over the response, printing each update to the console.
+
+#### Cache responses
+
+If you're familiar with [Caching in .NET](caching.md), it's good to know that <xref:Microsoft.Extensions.AI> provides other such delegating `IChatClient` implementations. The <xref:Microsoft.Extensions.AI.DistributedCachingChatClient> is an `IChatClient` that layers caching around another arbitrary `IChatClient` instance. When a unique chat history is submitted to the `DistributedCachingChatClient`, it forwards it along to the underlying client, and then caches the response before it being sent back to the consumer. The next time the same history is submitted, such that a cached response can be found in the cache, the `DistributedCachingChatClient` can return back the cached response rather than needing to forward the request along the pipeline.
```

Comment:
```suggestion
If you're familiar with [Caching in .NET](caching.md), it's good to know that <xref:Microsoft.Extensions.AI> provides other such delegating `IChatClient` implementations. The <xref:Microsoft.Extensions.AI.DistributedCachingChatClient> is an `IChatClient` that layers caching around another arbitrary `IChatClient` instance. When a unique chat history is submitted to the `DistributedCachingChatClient`, it forwards it to the underlying client and then caches the response before sending it back to the consumer. The next time the same history is submitted, such that a cached response can be found in the cache, the `DistributedCachingChatClient` returns the cached response rather than needing to forward the request along the pipeline.
```